modules/generic_worker/manifests/init.pp

```diff
@@ -66,6 +79,72 @@
             }
         }
+        Ubuntu: {
+            case $::operatingsystemrelease {
+                16.04: {
+                    if ($environment == 'staging') {
+                        # We are limited to 22 characters for worker_type
+                        $worker_type = "gecko-t-linux-talos-b"
+                        $taskcluster_client_id = secret('generic_worker_linux_staging_client_id')
+                        $taskcluster_access_token = hiera('generic_worker_linux_staging_access_token')
+                    }
+                    else {
+                        $worker_type = "gecko-t-osx-linux-talos"
+                        $taskcluster_client_id = secret('generic_worker_linux_client_id')
+                        $taskcluster_access_token = hiera('generic_worker_linux_access_token')
+                    }
+                    file {
+                        ["${::users::builder::home}/.config",
+                        "${::users::builder::home}/.config/autostart"]:
+                            ensure => directory,
+                            owner  => $users::builder::username,
+                            group  => $users::builder::group;
+                        "${::users::builder::home}/.config/autostart/gnome-terminal.desktop":
+                            content => template('generic_worker/gnome-terminal.desktop.erb'),
+                            owner   => $users::builder::username,
+                            group   => $users::builder::group;
+                    }
+                    file { '/usr/local/bin/run-generic-worker.sh':
+                        ensure  => present,
+                        content => template('generic_worker/run-generic-worker.sh.erb'),
+                        mode    => '0755',
+                        owner   => $users::root::username,
+                        group   => $users::root::group;
+                    }
+                    file { '/etc/generic-worker.config':
+                        ensure  => present,
+                        content => template('generic_worker/generic-worker.config.erb'),
+                        mode    => '0644',
+                        owner   => $users::root::username,
+                        group   => $users::root::group;
+                    }
+                    exec { 'create gpg key':
+                        path    => ['/bin', '/sbin', '/usr/local/bin', '/usr/bin'],
+                        user    => 'cltbld',
+                        cwd     => $users::builder::home,
+                        command => "generic-worker new-openpgp-keypair --file ${::users::builder::home}/generic-worker.openpgp.key",
+                        unless  => "test -f ${::users::builder::home}/generic-worker.openpgp.key"
+                    }
+
+                    host {"${taskcluster_host}":
+                        ip => '127.0.0.1'
+                    }
+                    # Enable proxy_http_module on apache2
```

Comment:
Make sure you are explicitly including the httpd (apache) module if you are going to be calling a2enmod and notifying the service defined there.  You might even want to require it just to be on the safe side.